### ISSUE TITLE
Disambiguate RETURNDATA in CallFrame

### ISSUE THREAD

**Issue by JereSalo** [lambdaclass/lambda_ethereum_rust#926]
The callframe has an attribute named returndata that sometimes is used as the actual return data of the current call frame and other times it is used as the place to store the return data of the sub-context. This is ambiguous.
I think that the first usage would be the most appropriate for it.
Maybe the return data of the sub-context is just stored in memory and not as a variable in the callframe.

The case in which I see the returndata is being misused is after performing a generic call. We shouldn't set the return data of the current context after that, I think we should only store the return data of the sub-context in memory (We have offset and size). And the name of those attributes (offset and size) should be more clear, so that the programmer knows that they actually represent where the return data of the SUB-context is stored.

Other problem that comes with this is that in opcodes like RETURNDATASIZE and RETURNDATACOPY, that should get info about the return data of the sub-context, we are actually getting info about the return data of the current context, which is wrong.

The return data is used properly in opcodes like RETURN, it is set to the value to be returned.

To summarize:
- Return data should store the return data of the current context
- The return data of the sub-context should be stored somewhere in memory (or maybe we could store it in another variable)
- generic_call() is misusing return_data after executing a sub-context
- RETURNDATASIZE and RETURNDATACOPY are using returndata as if it was the return data of the sub-context, when it is not.

Maybe if we want to store it in a variable we could have something like `sub_return_data` and keep `return_data`.
